### PR TITLE
Raise error when missmatch between boot filename and registered component

### PR DESCRIPTION
**Pull request by GustavoCaso**
This solve #38 

@timriley, @solnic  I don't know if you have it in the roadmap or we do not want to raise such error. But I thought to give it a try since it looks quite simple to implement.

### CODE REVIEW

File: lib/dry/system/booter.rb [hunk 1]
```diff
@@ -66,11 +66,11 @@ def boot!(name)
       def call(name)
         container, finalizer = finalizers[name]
 
-        if finalizer
-          lifecycle = Lifecycle.new(container, &finalizer)
-          yield(lifecycle) if block_given?
-          lifecycle
-        end
+        raise MissmatchBetweenFileAndRegisteredComponents.new(name, registered_booted_keys) unless finalizer
```
Comment (solnic) [inline lib/dry/system/booter.rb]: We could probably use a shorter name, maybe `ComponentFileMismatchError`?

File: lib/dry/system/errors.rb [hunk 2]
```diff
@@ -10,6 +10,15 @@ def initialize(component)
       end
     end
 
+    # Error raised when booter file do not match with register component
+    #
+    # @api public
+    MissmatchBetweenFileAndRegisteredComponents = Class.new(StandardError) do
+      def initialize(filename, registered_booted_keys)
+        super("Missmatch between filename +#{filename}+ and registered components +#{registered_booted_keys}+")
```
Comment (solnic) [inline lib/dry/system/errors.rb]: typo `Missmatch` => `Mismatch`

**Comment by GustavoCaso** [comment]
@solnic much better and shorter name

File: spec/unit/container_spec.rb [hunk 3]
```diff
@@ -142,6 +142,14 @@ class Test::Container < Dry::System::Container
           'component identifier +foo+ is invalid or boot file is missing'
         )
       end
+
+      describe "missmatch betwenn finalize name and registerd component" do
```
Comment (solnic) [inline spec/unit/container_spec.rb]: Small typo `registerd` => `registered`

File: spec/unit/container_spec.rb [hunk 4]
```diff
@@ -142,6 +142,14 @@ class Test::Container < Dry::System::Container
...
+        it "raises MissmatchBetweenFileAndRegisteredComponents" do
```
Comment (solnic) [inline spec/unit/container_spec.rb]: You can just write `raises a meaningful error`

File: spec/unit/container_spec.rb (hunk 3 above)
Comment (GustavoCaso) [inline spec/unit/container_spec.rb]: I will change it right now

**Comment by GustavoCaso** [comment]
@solnic I have addressed the changes. Hope this is good to go.

**Pull request merged**